server/schedulers/balance_leader.go

```diff
@@ -364,6 +364,11 @@ func (l *balanceLeaderScheduler) Schedule(cluster schedule.Cluster, dryRun bool)
 
 	result := make([]*operator.Operator, 0, batch)
 	for sourceCandidate.hasStore() || targetCandidate.hasStore() {
+		// if coordinator is stopping, speed up exit.
+		if l.BaseScheduler.OpController.Ctx().Err() != nil {
```

Comment:
Maybe we can consider adding the `context` field for all schedulers?

Reply:
Every scheduler have a BaseScheduler, and BaseScheduler have opController with a context filed, I think it's no difference.